CRM/PivotReport/DataActivity.php

```diff
@@ -0,0 +1,374 @@
+<?php
+
+/**
+ * Provides a functionality to prepare entity data for Pivot Table.
+ */
+class CRM_PivotReport_DataActivity extends CRM_PivotReport_AbstractData {
+
+  public function __construct($name = NULL) {
+    $this->name = 'Activity';
+  }
+
+  /**
+   * Rebuilds pivot report cache including header and data.
+   *
+   * @param \CRM_PivotCache_AbstractGroup $cacheGroup
+   * @param array $params
+   *
+   * @return array
+   */
+  public function rebuildCache($cacheGroup, array $params) {
+    $this->fields = $this->getFields();
+    $this->emptyRow = $this->getEmptyRow();
+    $this->multiValues = array();
+
+    $time = microtime(true);
+
+    $cacheGroup->clear();
+
+    $count = $this->rebuildData($cacheGroup, $this->name, $params);
```

Comment:
`rebuildData` is an instance method and it has access to all of the instance property, so there's no need to pass `$this->name` as a param